tez-dag/src/main/java/org/apache/tez/dag/app/DAGAppMaster.java

```diff
@@ -766,6 +770,10 @@ protected synchronized void handle(DAGAppMasterEvent event) {
           "DAGAppMaster Internal Error occurred");
       break;
     case DAG_FINISHED:
+      if (tezThreadDumpHelper != null) {
+        tezThreadDumpHelper.shutdownPeriodicThreadDumpService();
```

Comment:
nit: is it possible that tezThreadDumpHelper wraps the actual class which is doing the actual work and you can simply tezThreadDumpHelper.start(conf), tezThreadDumpHelper.stop() it? in general I'm trying to avoid this "= null" boilerplate